Code/Editor/GameEngine.cpp

```diff
@@ -794,6 +795,22 @@ void CGameEngine::Update()
         {
             pRenderViewport->Update();
         }
+
+        // Check for the Escape key to exit game mode here rather than in Qt, 
+        // because all Qt events are usually filtered out in game mode in 
+        // QtEditorApplication_<platform>.cpp nativeEventFilter() to prevent 
+        // using Editor menu actions and shortcuts that shouldn't trigger while 
+        // playing the game.
+        // When the user opens the console, Qt events will be allowed 
+        // so the user can interact with limited Editor content like the console.
+        const AzFramework::InputChannel* inputChannel = nullptr;
+        const AzFramework::InputChannelId channelId(AzFramework::InputDeviceKeyboard::Key::Escape);
+        AzFramework::InputChannelRequestBus::EventResult(inputChannel, channelId, &AzFramework::InputChannelRequests::GetInputChannel);
+        if(inputChannel && inputChannel->GetState() == AzFramework::InputChannel::State::Began)
+        {
+            // leave game mode
+            RequestSetGameMode(false);
+        }
```

Comment:
Is the input only processed once per frame?

Because if not, could the input channel state go from Began ->Updated before it is processed?

Reply:
The input is only processed once per tick/frame at the beginning so we should always see the 'Began' state unless someone is doing something unusual with the input system.  I considered putting this block of code before the game ticks, but thought it made more sense to let the game application get to process the Escape key too before the Editor uses it to leave game mode.  Sometimes it's good to see your game code is able to detect the Escape key for testing even though in Editor it results in exiting game mode.  Ideally, we'd led the user choose the key bind for this instead of hardcoding it to Escape.